src/main/groovy/io/seqera/wave/service/blob/impl/BlobCacheServiceImpl.groovy

```diff
@@ -293,4 +317,45 @@ class BlobCacheServiceImpl implements BlobCacheService {
             }
         }
     }
+
+   /**
+    * get the size of the blob stored in the cache
+    *
+    * @return {@link Long} the size of the blob stored in the cache
+    */
+   protected Long getBlobSize(String key) {
+        try {
+            final headObjectRequest =
+                    HeadObjectRequest.builder()
+                            .bucket(blobConfig.storageBucket.replaceFirst("s3://",""))
+                            .key(key)
+                            .build()
+            final headObjectResponse = s3Client.headObject(headObjectRequest as HeadObjectRequest)
+            final contentLength = headObjectResponse.contentLength()
+            return contentLength ?: 0L
+        }
+        catch (Exception e){
+            log.error("== Blob cache Error getting content length of object $key from bucket ${blobConfig.storageBucket}", e)
+            return 0L
```

Comment:
Would not be better to return -1 in case of error ?

Reply:
its size, so i think it can't be negative.

Reply:
that's my point, since an error is reported it should be an invalid size

Reply:
ok understood, i will make the changes

Reply:
done